cdc/capture/capture.go

```diff
@@ -159,11 +175,32 @@ func (c *Capture) reset(ctx context.Context) error {
 	if c.grpcPool != nil {
 		c.grpcPool.Close()
 	}
+
+	if config.SchedulerV2Enabled {
+		c.grpcService.Reset(nil)
+
+		if c.MessageRouter != nil {
+			c.MessageRouter.Close()
+			c.MessageRouter.Wait()
```

Comment:
Can we wait  in `Close`?

Reply:
Separating `Close` and `Wait` in the design of the interface is for two reasons:
1. To provide a finer-grained control over the timing of execution to achieve better unit-test coverage.
2. To avoid introducing deadlock bugs accidentally in the future. The pattern of `close-wait-close-wait` instead of `close-close-wait-wait` has created deadlock bugs before.